hashmap/src/main.rs

```diff
@@ -0,0 +1,248 @@
+use std::collections::HashMap;
+
+//mod tej_serializer;
+pub trait MeasurementCollector {
```

Comment:
```rust

trait MeasurementVisitor {
    type Error;
    fn timestamp(&mut self, value: String) -> Result<(), Self::Error>;
    fn measurement(
        &mut self,
        group: Option<&str>,
        name: &str,
        value: f64,
    ) -> Result<(), Self::Error>;
}
```

Reply:
```rust

trait Visitor {
    type Error;
    type Item;
    fn visit(&mut self, item: Self::Item) -> Result<(), Self::Error>;
}
```

Reply:
```rust
trait FlatMeasurementVisitor {
    type Error;
    fn timestamp(&mut self, value: String) -> Result<(), Self::Error>;
    fn measurement(
        &mut self,
        group: Option<&str>,
        name: &str,
        value: f64,
    ) -> Result<(), Self::Error>;
}

trait GroupedMeasurementVisitor {
    type Error;
    fn timestamp(&mut self, value: String) -> Result<(), Self::Error>;
    fn measurement(
        &mut self,
        group: Option<&str>,
        name: &str,
        value: f64,
    ) -> Result<(), Self::Error>;
  fn start_group(&mut self, group: &str)  -> Result<(), Self::Error>;
  fn end_group(&mut self)  -> Result<(), Self::Error>;
}
```

Reply:
But this is like just further reducing the granularity of the measurement implementation. Whats the advantage?.